restli-tools/src/main/resources/apiVmTemplates/rest.create_return_entity.vm

```diff
@@ -0,0 +1,64 @@
+#*
+   Copyright (c) 2021 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*#
+
+  #doc($method.schema.doc)
+  public CompletionStage<IdEntityResponse<${spec.keyClassName}, ${spec.entityClassName}>> create(
+      ${spec.entityClassName} entity,
+      boolean returnEntity#if( $method.hasParams()),#end
+      #methodParams($method)
+      ) {
+    Map<String, Object> queryParams = new HashMap<>();
```

Comment:
There were somehow no indention for these lines of code generated in my machine. Is that different on your side?

Reply:
That is odd. I bumped velocity version specifically for this and the indentation looks okay on my machine. Did you run ./gradlew build?

Reply:
It is reproduced in at least two machines (I didn't use Mac)
![Screen Shot 2021-01-28 at 2 04 25 AM](https://user-images.githubusercontent.com/52180478/106122103-28afa180-610d-11eb-8818-172181571cb5.png)

Reply:
I also reproduced it. This happened after I moved some of the templates to macros. It looks like macros do not keep the source file indentation. 
The only option is to not use macros, but that would mean logic duplication.
I've edited the macros to indroduce some spaces based on where they are used (not perfect). I guess we'd have to live with this oddity..

Reply:
I have looked up some documentations and I believe it is tunable. But if it is hard to do so properly now or if it is not so important right now, can we mark it somewhere as a **TODO** item?

I tested with the following and it worked on my machine. It seems we need to both (a) add velocity comment indentation in the marco (b) Add indention through velocity comment when use it

Source: https://stackoverflow.com/questions/43304493/how-to-keep-template-whitespace-tabs-formatting-using-apache-velocity-1-7
(Didn't try with the space gobbling option)

![Screen Shot 2021-01-28 at 9 02 39 AM](https://user-images.githubusercontent.com/52180478/106172347-94aefb80-6147-11eb-89c5-6cf8e9fc7cfa.png)

Reply:
Thanks for trying out, I've made the change. This is still not ideal as the indentation is fixed (what if we use the macro at a different indentation level?)
There are definitely ways to do it, for example, you can pass in an $indent arg to the macro. But that is hard to maintain. I think this is a good trade-off for now.